lighthouse-core/gather/computed/new-computed-artifact.js

```diff
@@ -8,28 +8,33 @@
 const ArbitraryEqualityMap = require('../../lib/arbitrary-equality-map.js');
 
 /**
- * @template {string} N
- * @template A
- * @template R
- * @param {{name: N, compute_: (artifacts: A, context: LH.Audit.Context) => Promise<R>}} computableArtifact
- * @return {{name: N, request: (artifacts: A, context: LH.Audit.Context) => Promise<R>}}
+ * Decorate computableArtifact with a caching `request()` method which will
+ * automatically call `computableArtifact.compute_()` under the hood.
+ * @template {{name: string, compute_(artifacts: unknown, context: LH.Audit.Context): Promise<unknown>}} C
+ * @param {C} computableArtifact
```

Comment:
it was a nice dream to encapsulate computed artifacts like they were before this PR, leaving only a
```js
{name: string, request: (artifacts: Whatever, context: LH.Audit.Context): Promise<Yeah>}
```
but we access static methods on the computed artifact classes in a few different places and `LanternFirstCPUIdle` even extends `LanternInteractive`, so we really need this function to return a
```js
InputComputedArtifact & {request: (artifacts: Whatever, context: LH.Audit.Context): Promise<Yeah>}
```
which this change does. There's some limitations with current tsc (3.1) and trying to do this in jsdoc (e.g. you can constrain templates in jsdoc but you can't provide default values), but dong some of this explicitly (as the comment added below mentions) makes it work without being overly prescriptive or falling back to `any`.